pkg/dockerregistry/server/auth.go

```diff
@@ -167,16 +153,57 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 					challenge.err = err
 					return nil, challenge
 				}
-
-				return ctx, nil
```

Comment:
It doesn't cause any problems, but the way the registry is coded now, you'll either get pull, pull+push, or prune.

Also, the only place the user client is used any more is when a push happens and the target image stream doesn't exist, the registry uses the user's credentials to try to create the image stream.